src/realm/query_expression.hpp

```diff
@@ -2183,6 +2245,75 @@ class LinkCount : public Subexpr2<Int> {
     LinkMap m_link_map;
 };
 
+template <class oper, class TLeft>
+class SizeOperator : public Subexpr2<Int> {
+public:
+    SizeOperator(std::unique_ptr<TLeft> left)
+        : m_left(std::move(left))
+    {
+    }
+
+    SizeOperator(const SizeOperator& other, QueryNodeHandoverPatches* patches)
+        : m_left(other.m_left->clone(patches))
+    {
+    }
+
+    // See comment in base class
+    void set_base_table(const Table* table) override
+    {
+        m_left->set_base_table(table);
+    }
+
+    void verify_column() const override
+    {
+        m_left->verify_column();
+    }
+
+    // Recursively fetch tables of columns in expression tree. Used when user first builds a stand-alone expression
+    // and binds it to a Query at a later time
+    const Table* get_base_table() const override
+    {
+        return m_left->get_base_table();
+    }
+
+    // destination = operator(left)
+    void evaluate(size_t index, ValueBase& destination) override
+    {
+        Value<Int>* d = dynamic_cast<Value<Int>*>(&destination);
+        REALM_ASSERT(d);
+
+        Value<T> left;
+        m_left->evaluate(index, left);
+
+        size_t sz = left.m_values;
+        d->init(left.m_from_link_list, sz);
+
+        for (size_t i = 0; i < sz; i++) {
+            auto elem = left.m_storage.get(i);
+            if (!elem) {
+                d->m_storage.set(i, util::none);
+            }
+            else {
+                d->m_storage.set(i, oper()(*elem));
+            }
+        }
+    }
+
+    std::unique_ptr<Subexpr> clone(QueryNodeHandoverPatches* patches) const override
+    {
+        return make_subexpr<SizeOperator>(*this, patches);
+    }
+
+    void apply_handover_patch(QueryNodeHandoverPatches& patches, Group& group) override
+    {
+        m_left->apply_handover_patch(patches, group);
+    }
+
+private:
+    typedef typename oper::type T;
+    std::unique_ptr<TLeft> m_left;
```

Comment:
`m_left` doesn't seem like the right name for an operator that only takes one object.